packages/graphql-server-express/src/expressApollo.ts

```diff
@@ -42,33 +42,48 @@ export function graphqlExpress(options: GraphQLOptions | ExpressGraphQLOptionsFu
     }
 
     const formatErrorFn = optionsObject.formatError || graphql.formatError;
+    let requestPayload;
+
+    switch ( req.method ) {
+      case 'POST':
+        if ( !req.body ) {
+          res.statusCode = 500;
+          res.write('POST body missing. Did you forget "app.use(bodyParser.json())"?');
+          res.end();
+          return;
+        }
 
-    if (req.method !== 'POST') {
-      res.setHeader('Allow', 'POST');
-      res.statusCode = 405;
-      res.write('Apollo Server supports only POST requests.');
-      res.end();
-      return;
-    }
+        requestPayload = req.body;
+        break;
+      case 'GET':
+        if ( !req.query ) {
+          res.statusCode = 500;
```

Comment:
Yeah, I think 400 is better, because in the case of GET, it means a request was sent without a query, which is definitely not an internal server error. Although... that reminds me that we should also think about how prepared queries would work with GET. I definitely think they should be supported as well, so either this test is in the wrong place, or prepared queries should be a middleware that adds the query to the request before it hits graphql-server.